src/Compilers/CSharp/Test/Emit2/Semantics/ExperimentalAttributeTests.cs

```diff
@@ -0,0 +1,880 @@
+
+    [Fact]
+    public void EmptyUrlFormat()
+    {
+        var src = """
+C.M();
+
+[System.Diagnostics.CodeAnalysis.Experimental("DiagID1", UrlFormat = "")]
+class C
+{
+    public static void M() { }
+}
+""";
+        var comp = CreateCompilation(new[] { src, experimentalAttributeSrc });
+        comp.VerifyDiagnostics(
+            // 0.cs(1,1): warning DiagID1: 'C' is for evaluation purposes only and is subject to change or removal in future updates.
+            // C.M();
+            Diagnostic("DiagID1", "C").WithArguments("C").WithLocation(1, 1)
+            );
+
+        var diag = comp.GetDiagnostics().Single();
+        Assert.Equal("DiagID1", diag.Id);
+        Assert.Equal(ErrorCode.WRN_Experimental, (ErrorCode)diag.Code);
+        Assert.Equal("", diag.Descriptor.HelpLinkUri);
+    }
+
+    [Fact]
+    public void NullUrlFormat()
+    {
+        // We use a default help URL if the UrlFormat is improper
+        var src = """
+C.M();
+
+[System.Diagnostics.CodeAnalysis.Experimental("DiagID1", UrlFormat = null)]
+class C
+{
+    public static void M() { }
+}
+""";
+        var comp = CreateCompilation(new[] { src, experimentalAttributeSrc });
+        comp.VerifyDiagnostics(
+            // 0.cs(1,1): warning DiagID1: 'C' is for evaluation purposes only and is subject to change or removal in future updates.
+            // C.M();
+            Diagnostic("DiagID1", "C").WithArguments("C").WithLocation(1, 1)
+            );
+
+        var diag = comp.GetDiagnostics().Single();
+        Assert.Equal("DiagID1", diag.Id);
+        Assert.Equal(ErrorCode.WRN_Experimental, (ErrorCode)diag.Code);
+        Assert.Equal(DefaultHelpLinkUri, diag.Descriptor.HelpLinkUri);
+    }
+
+    [Fact]
+    public void NullUrlFormat_Metadata()
+    {
+        var libSrc = """
+[System.Diagnostics.CodeAnalysis.Experimental("DiagID", UrlFormat = null)]
+public class C
+{
+    public static void M() { }
+}
+
+namespace System.Diagnostics.CodeAnalysis
+{
+    [AttributeUsage(AttributeTargets.All, Inherited = false)]
+    public sealed class ExperimentalAttribute : Attribute
+    {
+        public ExperimentalAttribute(string diagnosticId) { }
+        public string UrlFormat { get; set; }
+    }
+}
+""";
+
+        var libComp = CreateCompilation(libSrc);
+
+        var src = """
+C.M();
+""";
+
+        var comp = CreateCompilation(src, references: new[] { libComp.EmitToImageReference() });
+        comp.VerifyDiagnostics(
+            // (1,1): warning DiagID: 'C' is for evaluation purposes only and is subject to change or removal in future updates.
+            // C.M();
+            Diagnostic("DiagID", "C").WithArguments("C").WithLocation(1, 1)
+            );
+    }
+
+    [Theory, CombinatorialData]
+    public void Suppressed(bool inSource)
+    {
+        var libSrc = """
+[System.Diagnostics.CodeAnalysis.Experimental("DiagID1")]
+public class C
+{
+    public static void M() { }
+}
+""";
+
+        var src = """
+#pragma warning disable DiagID1
```

Comment:
Consider testing suppression using `SuppressMessageAttribute`.

Reply:
Good idea. But it looks like SuppressMessageAttribute doesn't work for compiler warnings, only analyzer warnings.